### ISSUE TITLE
When adding Turn Restrictions, split the from and to ways near the intersection

### ISSUE THREAD

**Issue by bhousel** [openstreetmap/iD#5000]
This is a request from several teams working with turn restrictions (including Mapbox and others).  It would be useful to split turn restriction from/to ways near the turn restriction intersection.  

![split_at_junctions](https://user-images.githubusercontent.com/38784/38995116-31e3ae80-43b6-11e8-928f-53b9156964a7.png)

The benefits here include:
* limit the extent of the effects of the turn restriction
* better review of turn restrictions in tools like on OSM or OSMCha or JOSM
* avoid unnecessary modification of the turn restriction (e.g. a "new" TR may end up with version 2 just because a split happened far away after it was added).

For example:

It's pretty easy to tell what was changed here because the turn restriction is scoped tightly to the intersection:
<img width="500" alt="screenshot 2018-04-19 09 57 39" src="https://user-images.githubusercontent.com/38784/38996035-47b5d51a-43b8-11e8-9e01-cfc77be15c67.png">

Not so much for this one, which covers a large portion of Detroit:
<img width="500" alt="screenshot 2018-04-19 09 58 20" src="https://user-images.githubusercontent.com/38784/38996080-645b1392-43b8-11e8-922a-7bb8e598f7c0.png">

(Both changes were made with iD - the first example just happened to involve ways that were already split around there).

cc @maning @planemad

**Comment by iandees** [comment]
If iD splits the way to limit the scope of the turn restriction, won't that changeset's bounding box will still be big (because the split way will force the changeset bbox to include the original way)? Future changesets that modify the turn restriction will be smaller, of course.

**Comment by bhousel** [comment]
> If iD splits the way to limit the scope of the turn restriction, won't that changeset's bounding box will still be big (because the split way will force the changeset bbox to include the original way)? Future changesets that modify the turn restriction will be smaller, of course.

Yes, the bounds of the **changeset** will still be just as big - the extent of the change to the ways will be large - but the bounds around the **restriction** will be small.  

Mostly the thing we are trying to avoid is the Detroit screenshot above.  It was from OSMCha, but you would see the same thing in JOSM or on `openstreetmap.org/relation/` when looking at a turn restriction relation:

<img width="500" alt="screenshot 2018-04-19 10 29 11" src="https://user-images.githubusercontent.com/38784/38997935-8e1c6538-43bc-11e8-9705-6fce06373ad2.png">

https://www.openstreetmap.org/relation/8223161#map=13/42.3946/-82.9343

**Comment by iandees** [comment]
Yep, I'm just saying that the changeset bounding box (the Detroit screenshot above) would not change if iD split the way.

**Comment by mmd-osm** [comment]
I put up an Overpass query on our dev server to identify such super large TRs:

http://overpass-turbo.eu/s/y4X

Maybe that's useful to identify some interesting test cases. Sometimes a good place to split isn't just around the corner: https://www.openstreetmap.org/relation/5964280

**Comment by bhousel** [comment]
Yes, that's an excellent example @mmd-osm 😱 
I think choosing a place to split could be defined by some rule like 
* "at the next junction within 100m"
* "at a node around 100m away or"
* "if no node like that exists, create one"

**Comment by Nakaner** [comment]
I think that this adds an unnecessary change to the from/to way. Please keep in mind that splits and merges of objects are difficult to track with OSM API 0.6.
